packages/tupaia-web/src/features/Dashboard/Dashboard.tsx

```diff
@@ -148,6 +149,19 @@ export const Dashboard = () => {
     }
   }, [dashboardNotFound, defaultDashboardName]);
 
+  const getIsDrilldown = (code: DashboardItemType['code']) => {
+    return activeDashboard?.items?.some(item => {
+      const { config } = item as {
+        config: MatrixConfig;
+      };
+      if (config?.drillDown && config?.drillDown?.itemCode === code) return true;
+      return false;
+    });
+  };
+
+  const visibleDashboards =
+    activeDashboard?.items?.filter(item => !getIsDrilldown(item.code)) || [];
```

Comment:
```suggestion
const visibleDashboards = activeDashboard?.items?.reduce((items , item) => {
  const isDrillDown = activeDashboard?.items?.some(({ config }) =>
    config?.drillDown && config?.drillDown?.itemCode === item.code
  );
  return isDrillDown ? items : [...items, item];
}, []);
```
this is an optional suggestion and pretty subjective change so it's up to you but I like using reduce for this since the activeDashboard items are used in the filter and the some

Reply:
ps. it would also be nice to add a short comment